OpenRA.Mods.Common/Widgets/Logic/Ingame/LeaveMapLogic.cs

```diff
@@ -33,6 +34,11 @@ public LeaveMapLogic(Widget widget, World world, OrderManager orderManager)
 		{
 			this.orderManager = orderManager;
 
+			if (world.LocalPlayer.WinState == WinState.Won)
+			{
+				CampaignProgress.SaveProgress(world.LocalPlayer.Country.Name);
```

Comment:
Indeed, but I meant that this adds extra complexity to the (already messy) end-game plumbing that can be avoided by using a slightly different abstraction.